provider/kubernetes/kubernetes.go

```diff
@@ -400,6 +405,15 @@ func getRuleForPath(pa extensionsv1beta1.HTTPIngressPath, i *extensionsv1beta1.I
 	}
 
 	ruleType := getStringValue(i.Annotations, annotationKubernetesRuleType, ruleTypePathPrefix)
+
+	switch ruleType {
+	case ruleTypePath, ruleTypePathPrefix, ruleTypePathStrip, ruleTypePathPrefixStrip:
+	case ruleTypeReplacePath:
+		log.Warnf("Using %s as %s will be deprecated in the future. Please use the %s annotation instead", ruleType, annotationKubernetesRuleType, annotationKubernetesRequestModifier)
+	default:
+		return "", fmt.Errorf("cannot use nonexistent, or non-matcher rule: %q", ruleType)
```

Comment:
How about adding a `case "":` and log the non-existent case separately? This would be slightly clearer as we don't have to log an empty string.